lib/iris/experimental/regrid.py

```diff
@@ -473,140 +473,251 @@ def _regrid_area_weighted_array(
-                new_data[tuple(indices)] = ma.masked
+    axis = (y_dim, x_dim)
+
+    # Stack the src_area data and weights for each target point
+    target_pt_ji = -1
+    for j, y_indices in enumerate(cached_y_indices):
+        for i, x_indices in enumerate(cached_x_indices):
+            target_pt_ji += 1
+            # Determine whether to mask element i, j based on whether
+            # there are valid weights.
+            weights = cached_weights[j][i]
+            if isinstance(weights, bool) and not weights:
+                if not src_masked:
+                    # Cheat! Fill the data with zeros and weights as one.
+                    # The weighted average result will be the same, but
+                    # we avoid dividing by zero.
+                    src_area_weights[..., target_pt_ji] = 1
+                    new_data_mask[..., j, i] = True
             else:
                 # Calculate weighted mean of data points.
                 # Slice out relevant data (this may or may not be a view()
                 # depending on x_indices being a slice or not).
-                if isinstance(x_indices, tuple) and isinstance(
-                    y_indices, tuple
-                ):
-                    raise RuntimeError(
-                        "Cannot handle split bounds " "in both x and y."
-                    )
-                # Calculate weights based on areas of cropped bounds.
-                weights = area_func(y_bounds, x_bounds)
-
-                if x_dim is not None:
-                    indices[x_dim] = x_indices
-                if y_dim is not None:
-                    indices[y_dim] = y_indices
-                data = src_data[tuple(indices)]
-
-                # Transpose weights to match dim ordering in data.
-                weights_shape_y = weights.shape[0]
-                weights_shape_x = weights.shape[1]
-                if x_dim is not None and y_dim is not None and x_dim < y_dim:
-                    weights = weights.T
-                # Broadcast the weights array to allow numpy's ma.average
-                # to be called.
-                weights_padded_shape = [1] * data.ndim
-                if y_dim is not None:
-                    weights_padded_shape[y_dim] = weights_shape_y
-                if x_dim is not None:
-                    weights_padded_shape[x_dim] = weights_shape_x
-                # Assign new shape to raise error on copy.
-                weights.shape = weights_padded_shape
-                # Broadcast weights to match shape of data.
-                _, weights = np.broadcast_arrays(data, weights)
-
-                # Calculate weighted mean taking into account missing data.
-                new_data_pt = _weighted_mean_with_mdtol(
-                    data, weights=weights, axis=axis, mdtol=mdtol
-                )
+                data = src_data[..., y_indices, x_indices]
+                Nx = data.shape[-1]
+                Ny = data.shape[-2]
```

Comment:
`len_x` and `len_y` rather than `Nx` and `Ny`?

Reply:
Done